cnf-tests/testsuites/e2esuite/nmstate/nmstate_sriov.go

```diff
@@ -0,0 +1,200 @@
+		Expect(err).ToNot(HaveOccurred())
+		DeferCleanup(client.Client.Delete, context.Background(), &nmstate)
+	})
+
+	Context("when a SriovNetworkNodePolicy and NodeNetworkConfigurationPolicySpec target the same NIC", func() {
+
+		BeforeAll(func() {
+			namespaces.CleanPods(testNamespace, client.Client)
+		})
+
+		It("VFs should spawn correctly", func() {
+			node, devices, err := findNonPrimarySriovDevicesAndNode(sriovInfos)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(devices)).ToNot(BeZero())
+
+			By("Using node " + node)
+
+			testDevice := devices[0]
+			By("Using device " + testDevice.Name)
+
+			By("Configuring IPAM on Physical Function via NMState")
+			nodeNetworkConfigPolicy := nmstatev1.NodeNetworkConfigurationPolicy{
+				ObjectMeta: metav1.ObjectMeta{Name: "test-sriov-device-" + testDevice.Name},
+				Spec: nmstateshared.NodeNetworkConfigurationPolicySpec{
+					NodeSelector: map[string]string{
+						"kubernetes.io/hostname": node,
+					},
+					DesiredState: nmstateshared.State{
+						Raw: nmstateshared.RawState(fmt.Sprintf(nmstateDesiredStateTemplateStaticIP, testDevice.Name)),
+					},
+				},
+			}
+			err = client.Client.Create(context.Background(), &nodeNetworkConfigPolicy)
+			Expect(err).ToNot(HaveOccurred())
+			DeferCleanup(client.Client.Delete, context.Background(), &nodeNetworkConfigPolicy)
+
+			waitForNMStatePolicyToBeStable(&nodeNetworkConfigPolicy)
+
+			By("Configuring VF on device via SriovNetworkNodePolicy")
+			sriovNetworkNodePolicy, err := sriovnetwork.CreateSriovPolicy(sriovclient, "test-nmstate-", namespaces.SRIOVOperator, testDevice.Name, node, 10, "testnmstateresource", "netdevice")
+			Expect(err).ToNot(HaveOccurred())
+			DeferCleanup(sriovclient.Delete, context.Background(), sriovNetworkNodePolicy)
+
+			Eventually(func() int64 {
+				testedNode, err := sriovclient.Nodes().Get(context.Background(), node, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				resNum := testedNode.Status.Allocatable[corev1.ResourceName("openshift.io/testnmstateresource")]
+				capacity, _ := resNum.AsInt64()
+				return capacity
+			}, 10*time.Minute, time.Second).Should(Equal(int64(10)))
+		})
+	})
+})
+
+func findNonPrimarySriovDevicesAndNode(n *cluster.EnabledNodes) (string, []*sriovv1.InterfaceExt, error) {
+	errs := []error{}
+
+	retNode := ""
+	retDevices := []*sriovv1.InterfaceExt{}
+
+	for _, node := range n.Nodes {
+		devices, err := n.FindSriovDevices(node)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		mainDeviceName, err := getPrimaryNICForNode(node)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		nonPrimaryDevices := []*sriovv1.InterfaceExt{}
+		for _, device := range devices {
+			if device.Name == mainDeviceName {
+				continue
+			}
+			nonPrimaryDevices = append(nonPrimaryDevices, device)
+		}
+
+		if len(nonPrimaryDevices) > len(retDevices) {
+			retNode = node
+			retDevices = nonPrimaryDevices
```

Comment:
Can you just return here in the happy path, and return the wrap errors.  I might be missing if there is selection between two nodes that have sriov or not

Reply:
good point: this function is not very clear. The point is to find the node with the most number of SR-IOV devices. Going to reword the variables and comment to make it clearer.